pkg/manifest/manifest.go

```diff
@@ -99,22 +148,77 @@ func ManifestsFromFiles(files []string) ([]Manifest, error) {
 }
 
 // ParseManifests parses a YAML or JSON document that may contain one or more
-// kubernetes resources.
-func ParseManifests(r io.Reader) ([]Manifest, error) {
+// kubernetes resources. Parsing and structure related errors are returned in
+// the first error parameter and cause an immediate return. Duplicate resource
+// errors are returned in the error list. Processing continues after a
+// duplicate resource error allowing all duplicates to be identified in the
+// given input.
+func ParseManifests(r io.Reader) ([]Manifest, error, []error) {
```

Comment:
I'm not sure we want the separate return value just for duplicates.  Failing fast if/when we detect duplicates might miss some entries if we are lousy with dups, but ideally the only dups are those in the in-flight PR being tested.

Reply:
Yeah, we're not losing much by not doing it and it makes it much easier to understand. Done.